### PR TITLE
[Messenger] Autoconfigurable attributes

### PR DESCRIPTION
**Pull request by alirezamirsepassi**
https://github.com/symfony/symfony/pull/43588

### CODE REVIEW

**Comment by alirezamirsepassi** [comment]
Any update on this PR ? @nicolas-grekas

**Review by 94noni — APPROVED** [review]

**Comment by 94noni** [comment]
Maybe just document that attributes requires php v8 (the feature is on v5.4 thus v8 is not mandatory like v6)

File: messenger.rst [hunk 1]
```diff
@@ -51,17 +51,19 @@ serialized::
 .. _messenger-handler:
 
 A message handler is a PHP callable, the recommended way to create it is to
-create a class that implements :class:`Symfony\\Component\\Messenger\\Handler\\MessageHandlerInterface`
-and has an ``__invoke()`` method that's type-hinted with the message class (or a
-message interface)::
+create a class using :class:`Symfony\\Component\\Messenger\\Attribute\\AsMessageHandler`
```
Comment (nicolas-grekas) [inline messenger.rst]: create a class that has the

File: messenger.rst [hunk 2]
```diff
@@ -51,17 +51,19 @@ serialized::
...
+attribute which has an ``__invoke()`` method that's type-hinted with the
```
Comment (nicolas-grekas) [inline messenger.rst]: and has an

File: messenger.rst [hunk 3]
```diff
@@ -51,17 +51,19 @@ serialized::
...
+message class (or a message interface) or you can create a class without the attribute, by implementing
```
Comment (nicolas-grekas) [inline messenger.rst]: I'd remove the mention of the interface

File: messenger.rst [hunk 4]
```diff
@@ -1769,6 +1772,36 @@ Customizing Handlers
 
 .. _messenger-handler-config:
 
+Configuring Handlers Using Attribute
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can configure your handler easily by passing options to the attribute::
```
Comment (nicolas-grekas) [inline messenger.rst]: "easily" is subjective, it should be removed

File: messenger.rst [hunk 5]
```diff
@@ -1769,6 +1772,36 @@ Customizing Handlers
 
 .. _messenger-handler-config:
 
+Configuring Handlers Using Attribute
```
Comment (nicolas-grekas) [inline messenger.rst]: Using Attributes (missing "s")

**Review by nicolas-grekas — COMMENTED** [review]
this should target 6.0 to me

**Comment by wouterj** [comment]
Thank you @alirezamirsepassi for providing docs with your code contributions.

I've applied all suggestions made by @nicolas-grekas and added some directives indicating this is a new feature. I did however merge it in 5.4, as that's where this was introduced (which is similar to other PHP attribute we've documented for 5.3/5.4).

**Pull request merged**